### PR TITLE
Removed URL from public API

### LINKED ISSUE #5656: Investigate if storage methods taking URL input should change to taking String url input
Currently, both the .NET and Java team are using URL type input for methods like `copyFromUrlWithResponse` and such. 
Investigate if they should be taking  String input instead.

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #5656 

Removed URL and URI as parameters from public client API, parameters now use Strings.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java [hunk 1]
```diff
@@ -395,8 +404,15 @@ public boolean isSnapshot() {
             .setSourceIfMatch(sourceModifiedAccessConditions.getIfMatch())
             .setSourceIfNoneMatch(sourceModifiedAccessConditions.getIfNoneMatch());
 
+        URL url;
+        try {
+            url = new URL(copySource);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(ex));
```
Comment (conniey) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java]: Should we add a message here with that malformed url as an inner exception.
("'copysource' is not a valid url.", ex)

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobAsyncClient.java [hunk 2]
```diff
@@ -205,21 +209,28 @@
...
         sourceRange = sourceRange == null ? new BlobRange(0) : sourceRange;
         destAccessConditions = destAccessConditions == null
             ? new AppendBlobAccessConditions() : destAccessConditions;
 
-        return this.azureBlobStorage.appendBlobs().appendBlockFromUrlWithRestResponseAsync(null, null, sourceURL, 0,
+        URL url;
+        try {
+            url = new URL(sourceUrl);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(ex));
```
Comment (conniey) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobAsyncClient.java]: I'd prefer to add a message to this, too. "'sourceUrl' is not a valid url.", Ex)

Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobAsyncClient.java]: Done

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java]: Done

**Pull request merged**